eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/diagram/compositereference/SCADiagramUpdateCompositeReferenceFeature.java

```diff
@@ -172,6 +172,14 @@ public boolean update(IUpdateContext context) {
                 for (ComponentReference componentReference : component.getReference()) {
                     if (reference.getName().equals(componentReference.getName())) {
                         promotedReferences.add(componentReference);
+
+                        // fix the promote issue if we can here for SWITCHYARD-2836
+                        if (reference.getPromote().isEmpty()) {
+                            if (reference.getInterface().eClass().getName().
```

Comment:
not in the SwitchYard editor. they would have to open the config file in the text or XML editor and edit it by hand, which to my mind invalidates any security they would have gotten from editing via the SY editor. The source tab in the SY editor is read only so they are unable to modify it there